sycl/source/detail/queue_impl.cpp

```diff
@@ -115,6 +115,22 @@ void queue_impl::addEvent(const event &Event) {
 /// addSharedEvent will have the queue track the events via a shared pointer.
 void queue_impl::addSharedEvent(const event &Event) {
   std::lock_guard<mutex_class> Lock(MMutex);
+  // Events stored in MEventsShared are not released anywhere else aside from
+  // calls to queue::wait/wait_and_throw, which a user application might not
+  // make, and ~queue_impl(). If the number of events grows large enough,
+  // there's a good chance that most of them are already completed and ownership
+  // of them can be released.
+  const size_t EventThreshold = 128;
```

Comment:
Is it possible to check that event_impl.MCommand is nullptr as an indication of that the command is completed?